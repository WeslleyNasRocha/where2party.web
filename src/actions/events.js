import { toast } from 'react-toastify';
import database from '../firebase/firebase';

export const addEvent = event => ({
  type: 'ADD_EVENT',
  event,
});

export const startAddEvent = (eventData = {}) => (dispatch) => {
  // const {
  //   descricao, titulo, local, data, image, orgId, address,
  // } = eventData;
  // const event = {
  //   descricao,
  //   titulo,
  //   local,
  //   data,
  //   image,
  //   orgId,
  //   address,
  // };
  const {
    Titulo, Descricao, Address, Data, Local,
  } = eventData;
  const event = {
    Titulo,
    Descricao,
    Address,
    Data,
    Local,
  };
  console.log(eventData);
  database
    .ref('eventos')
    .push(event)
    .then((ref) => {
      toast.success('Evento Adicionado com sucesso');
      dispatch(addEvent({
        id: ref.key,
        ...event,
      }));
    });
};

export const removeEvent = ({ id } = {}) => ({
  type: 'REMOVE_EVENT',
  id,
});

export const editEvent = (id, updates) => ({
  type: 'EDIT_EVENT',
  id,
  updates,
});

export const setEvents = events => ({
  type: 'SET_EVENTS',
  events,
});

export const startSetEvent = () => dispatch =>
  database
    .ref('eventos')
    .once('value')
    .then((snapshot) => {
      const events = [];

      snapshot.forEach((childSnapshot) => {
        events.push({
          id: childSnapshot.key,
          ...childSnapshot.val(),
        });
      });

      dispatch(setEvents(events));
    });
