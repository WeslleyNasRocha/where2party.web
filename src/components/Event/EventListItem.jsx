import React from 'react';
import { Link } from 'react-router-dom';

export const EventListItem = ({
  id, Titulo, Descricao, Address, Data, Local,
}) => (
  <div>
    <div className="box box-widget widget-user">
      {/* <Link to={`/edit/${id}`}> */}
      <Link to={`/edit/${id}`}>
        <h3>{Titulo}</h3>
      </Link>
      {/* </Link> */}
      <p>Descricao: {Descricao}</p>
      <p>Data: {Data}</p>
      <p>Address: {Address}</p>
      <p>
        Local: <b>long: </b>
        {Local.longitude} <b>lat: </b>
        {Local.latitude}
      </p>
    </div>
  </div>
);

export default EventListItem;
