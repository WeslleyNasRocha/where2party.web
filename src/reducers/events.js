const eventReducerDefaultState = [];

const eventReducer = (state = eventReducerDefaultState, action) => {
  // console.log(action.type, action);
  switch (action.type) {
    case 'ADD_EVENT':
      return [...state, action.event];
    case 'REMOVE_EVENT':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_EVENT':
      return state.map((event) => {
        if (event.id === action.id) {
          return {
            ...event,
            ...action.updates,
          };
        }
        return event;
      });
    case 'SET_EVENTS':
      return action.events;
    // return ['carai'];
    default:
      return state;
  }
};

export default eventReducer;
