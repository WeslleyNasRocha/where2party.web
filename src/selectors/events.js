import moment from 'moment';

// Get visible events

export default (events, {
  text, startDate, endDate,
}) =>
  events
    .filter((event) => {
      const dataEvent = moment(event.data, 'DD/MM/YY');
      const createdAtMoment = moment(dataEvent);
      const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
      const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
      const textMatch = event.titulo.toLowerCase().includes(text.toLowerCase());

      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => (a.createdAt < b.createdAt ? 1 : -1));
