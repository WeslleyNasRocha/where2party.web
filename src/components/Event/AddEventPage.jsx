import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import EventsForm from './EventsForm';
import { startAddEvent, addEvent } from '../../actions/events';

export class AddEventPage extends Component {
  onSubmit = (event) => {
    this.props.startAddEvent(event);
    // .then(() => {
    //   // this.props.history.push('/events');
    //   console.log('foi');
    // })
    // .catch(e => console.log(e));
  };
  render() {
    return (
      <div className="col-md-6">
        <div className="box box-success">
          <div className="box-header with-border">
            <h2 className="box-title">Add Event Page</h2>
          </div>
          <EventsForm submit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddEvent: event => dispatch(startAddEvent(event)),
});

export default connect(undefined, mapDispatchToProps)(AddEventPage);
