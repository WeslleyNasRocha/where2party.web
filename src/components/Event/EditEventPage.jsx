import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import EventsForm from './EventsForm';

import { startEditEvent, startRemoveEvent } from '../../actions/events';

export class EditEventPage extends Component {
  onSubmit = (event) => {
    // console.log(event);
    this.props.startEditEvent(event, this.props.evento.id);
    this.props.history.push('/events');
  };
  onRemove = () => {
    this.props.startRemoveEvent(this.props.evento);
    this.props.history.push('/events');
  };
  render() {
    return (
      <div className="col-md-10">
        <div className="box box-warning">
          <div className="box-header with-border">
            <h2 className="box-title">Editar Evento</h2>
          </div>
          <EventsForm submit={this.onSubmit} evento={this.props.evento} remove={this.onRemove} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startEditEvent: (updates, id) => dispatch(startEditEvent(updates, id)),
  startRemoveEvent: ({ id }) => dispatch(startRemoveEvent({ id })),
});

const mapStateToProps = (state, props) => ({
  evento: state.events.find(evento => evento.id === props.match.params.id),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditEventPage);
