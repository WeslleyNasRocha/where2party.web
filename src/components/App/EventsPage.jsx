import React from 'react';
import { connect } from 'react-redux';

import EventListItem from './EventListItem';

const EventsPage = props => (
  <div>
    {console.log(props.events)}
    {props.events.lenght === 0 ? (
      <p>sem eventos</p>
    ) : (
      props.events.map(event => <EventListItem key={event.id} {...event} />)
    )}
  </div>
);

const mapStateToProps = state => ({
  events: state.events,
});

export default connect(mapStateToProps)(EventsPage);
