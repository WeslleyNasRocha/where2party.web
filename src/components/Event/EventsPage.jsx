import React from 'react';
import { connect } from 'react-redux';

import EventListItem from './EventListItem';

export const EventsPage = props => (
  <div>
    {props.events.length !== 0 ? (
      props.events.map(event => <EventListItem key={event.id} {...event} />)
    ) : (
      <p>sem eventos</p>
    )}
  </div>
);

const mapStateToProps = (state) => {
  console.log(state);
  return {
    events: state.events,
  };
};

export default connect(mapStateToProps)(EventsPage);
