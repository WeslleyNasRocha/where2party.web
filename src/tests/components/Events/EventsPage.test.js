import React from 'react';
import { shallow } from 'enzyme';
import { EventsPage } from '../../../components/Event/EventsPage';

import events from '../../fixtures/events';

test('should render eventspage with events', () => {
  const wrapper = shallow(<EventsPage events={events} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render eventspage without events', () => {
  const wrapper = shallow(<EventsPage events={[]} />);
  expect(wrapper).toMatchSnapshot();
});
