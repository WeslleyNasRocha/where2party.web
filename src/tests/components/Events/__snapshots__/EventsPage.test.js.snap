// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render eventspage with events 1`] = `
<div>
  <EventListItem
    address="Av. Comendador Enzo Ferrari, 228 - Swift, Campinas - SP, Brazil"
    data="10/01/17"
    descricao="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vulputate mi in turpis dapibus facilisis."
    id="1"
    image="X2e8wZADnwhU3EonNmH6ppqO9g73++1505237223764.jpeg"
    key="1"
    local={
      Object {
        "latitude": -22.925993333333334,
        "latitudeDelta": 0.015,
        "longitude": -47.03723,
        "longitudeDelta": 0.0121,
      }
    }
    orgId="X2e8wZADnwhU3EonNmH6ppqO9g73"
    titulo="Chacrinha"
  />
  <EventListItem
    address="Av. Comendador Enzo Ferrari, 228 - Swift, Campinas - SP, Brazil"
    data="30/12/17"
    descricao="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vulputate mi in turpis dapibus facilisis."
    id="2"
    image="X2e8wZADnwhU3EonNmH6ppqO9g73++1505237223764.jpeg"
    key="2"
    local={
      Object {
        "latitude": -22.925993333333334,
        "latitudeDelta": 0.015,
        "longitude": -47.03723,
        "longitudeDelta": 0.0121,
      }
    }
    orgId="X2e8wZADnwhU3EonNmH6ppqO9g73"
    titulo="RolezÃ£o"
  />
  <EventListItem
    address="Av. Comendador Enzo Ferrari, 228 - Swift, Campinas - SP, Brazil"
    data="21/11/18"
    descricao="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vulputate mi in turpis dapibus facilisis."
    id="3"
    image="X2e8wZADnwhU3EonNmH6ppqO9g73++1505237223764.jpeg"
    key="3"
    local={
      Object {
        "latitude": -22.925993333333334,
        "latitudeDelta": 0.015,
        "longitude": -47.03723,
        "longitudeDelta": 0.0121,
      }
    }
    orgId="X2e8wZADnwhU3EonNmH6ppqO9g73"
    titulo="Festa na piscina"
  />
</div>
`;

exports[`should render eventspage without events 1`] = `
<div>
  <p>
    sem eventos
  </p>
</div>
`;
